components/layermanager/layermanager.service.ts

```diff
@@ -789,6 +791,7 @@ export class HsLayerManagerService {
       layer.layer.checkedSubLayers = {};
       layer.layer.withChildren = {};
     }
+    this.HsLayerSelectorService.layerSelected.next(layer);
```

Comment:
this.HsLayerSelectorService.select(layer) would be better I think. From what I understood from the article it's a good practice to hide the subject as a private variable and issue the next() from a public function inside the service (select() in our case) so outside services could not issue the events uncontrolled and create 'event soup'.

Reply:
Indeed, this way I completely left the select method out , not sure why. thanks for pointing it out.